### PR TITLE
Create KPrepublic directory

### PR DESCRIPTION
**Pull request by zvecr**
## Description

To reduce clutter in the `keyboards/` directory, this PR adds a `kprepublic` folder to collate all boards under one parent.

Should move as they currently mention kprepublic:
- cospad
- daisy
- staryu
- jj40
- jj50
- jj4x4
- xd84
- xd87

**Note:** I will do a 2nd pass and refactor each board to the current standards, including use of i2c as a separate set of PRs.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mechmerlin** [comment]
I'm a little concerned about putting in XD boards into this mix as I'm not too clear on whether XD boards are MADE FOR Kprepublic or whether Kprepublic is just selling XD boards.

**Comment by zvecr** [comment]
I agree, its a bit ambiguous with a lot of the currently supported boards, vendor vs reseller. Happy to make any changes as required. Figured at least here its a constant thread, instead of it getting lost on discord.

File: keyboards/kprepublic/cospad/readme.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
-COSPAD
-===
+#COSPAD
```
Comment (noroadsleft) [inline keyboards/kprepublic/cospad/readme.md]: Need a space after the `#` for this to work.

```suggestion
# COSPAD
```

File: keyboards/kprepublic/cospad/readme.md [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+
+![COSPAD](https://ae01.alicdn.com/kf/HTB1.lwARFXXXXcKXXXXq6xXFXXXK.jpg?size=289297&height=518&width=750&hash=7c5d826f928b91436cb36dca2746b424)
 
 A DIY Keypad Kit sold by KPRepublic, runs TKG natively.
 
 Keyboard Maintainer: QMK Community  
 Hardware Supported: COSPAD  
-Hardware Availability: [KPrepublic on Aliexpress](https://aliexpress.com/item/cospad-Custom-Mechanical-Keyboard-Kit-up-tp-24-keys-Supports-TKG-TOOLS-Underglow-RGB-PCB-20/32818383873.html)
+Hardware Availability: [KPrepublic](https://kprepublic.com/collections/cospad-xd24/products/cospad-20-numpad-xd24-xiudi-custom-keyboard-pcb)
```
Comment (noroadsleft) [inline keyboards/kprepublic/cospad/readme.md]: I wouldn't call this *wrong*, but the AliExpress link is still valid.

I'd prefer including both, and then anyone interested in this board can use the purchase link that works best for them.

```suggestion
Hardware Availability: [KPrepublic](https://kprepublic.com/collections/cospad-xd24/products/cospad-20-numpad-xd24-xiudi-custom-keyboard-pcb); [AliExpress](https://aliexpress.com/item/cospad-Custom-Mechanical-Keyboard-Kit-up-tp-24-keys-Supports-TKG-TOOLS-Underglow-RGB-PCB-20/32818383873.html)
```

File: keyboards/kprepublic/daisy/readme.md [hunk 3]
```diff
@@ -1,14 +1,16 @@
 # Daisy
 
+![Daisy](https://cdn.shopify.com/s/files/1/2711/4238/products/HTB1ahNecamgSKJjSsplq6yICpXaY_1024x1024.jpg?v=1515526190)
+
 A 40% keyboard kit sold by KPRepublic.
 
 Keyboard Maintainer: westfoxtrot (https://github.com/westfoxtrot)  
 Hardware Supported: Daisy PCB Rev.1, Daisy PCB Rev.2  
-Hardware Availability: http://tinyurl.com/yc26lq22
+Hardware Availability: [KPrepublic](https://kprepublic.com/collections/daisy-41/products/daisy-40-custom-keyboard-pcb)
```
Comment (noroadsleft) [inline keyboards/kprepublic/daisy/readme.md]: Ditto the above.

```suggestion
Hardware Availability: [KPrepublic](https://kprepublic.com/collections/daisy-41/products/daisy-40-custom-keyboard-pcb); [AliExpress](https://kprepublic.aliexpress.com/store/group/Daisy-40/3034003_512069836.html)
```

File: keyboards/kprepublic/daisy/readme.md [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
 
 Make example for this keyboard (after setting up your build environment):
 
-    make daisy:default
+    make kprepublic/daisy:default
 
 See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (noroadsleft) [inline keyboards/kprepublic/daisy/readme.md]: Can you please update this?

```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/kprepublic/jj40/readme.md [hunk 5]
```diff
@@ -6,11 +6,11 @@ A compact 40% (12x4) ortholinear keyboard kit made and KPRepublic on AliExpress.
 
 Keyboard Maintainer: [QMK Community](https://github.com/qmk)  
 Hardware Supported: Atmega32A  
-Hardware Availability: [AliExpress](https://www.aliexpress.com/store/product/jj40-Custom-Mechanical-Keyboard-40-PCB-programmed-40-planck-layouts-bface-firmware-gh40/3034003_32828781103.html)
+Hardware Availability: [KPrepublic](https://kprepublic.com/collections/pcb/products/jj40-v1-5-40-custom-keyboard-pcb-similar-with-planck)
```
Comment (noroadsleft) [inline keyboards/kprepublic/jj40/readme.md]:
```suggestion
Hardware Availability: [KPrepublic](https://kprepublic.com/collections/pcb/products/jj40-v1-5-40-custom-keyboard-pcb-similar-with-planck); [AliExpress](https://www.aliexpress.com/store/product/jj40-Custom-Mechanical-Keyboard-40-PCB-programmed-40-planck-layouts-bface-firmware-gh40/3034003_32828781103.html)
```

File: keyboards/kprepublic/jj40/readme.md [hunk 6]
```diff
@@ -6,11 +6,11 @@ A compact 40% (12x4) ortholinear keyboard kit made and KPRepublic on AliExpress.
...
 
 Make example for this keyboard (after setting up your build environment):
 
-    make jj40:default:program
+    make kprepublic/jj40:default:program
 
 See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (noroadsleft) [inline keyboards/kprepublic/jj40/readme.md]: Can you please update this?

```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/kprepublic/jj4x4/readme.md [hunk 7]
```diff
@@ -6,11 +6,11 @@ A  4x4 keypad kit made and KPRepublic on AliExpress. This is a chopped off versi
 
 Keyboard Maintainer: [QMK Community](https://github.com/qmk)  
 Hardware Supported: Atmega32A  
-Hardware Availability: [AliExpress](https://www.aliexpress.com/item/jj4x4-jj4X4-16-keys-Custom-Mechanical-Keyboard-PCB-programmed-numpad-layouts-bface-firmware-with-rgb-bottom/32901955446.html?spm=2114.search0104.3.7.3ebf431ae1d9ic&ws_ab_test=searchweb0_0,searchweb201602_4_10065_10130_10068_10547_319_317_10548_10545_10696_453_10084_454_10083_433_10618_431_10307_537_536_10902_10059_10884_10887_321_322_10103,searchweb201603_6,ppcSwitch_0&algo_expid=9d1891dd-80af-4793-a889-5a62e1fdfdd8-1&algo_pvid=9d1891dd-80af-4793-a889-5a62e1fdfdd8&transAbTest=ae803_5)
+Hardware Availability: [KPrepublic](https://kprepublic.com/products/jj4x4-jj4x4-16-keys-custom-mechanical-keyboard-pcb-programmed-numpad-layouts-bface-firmware-with-rgb-bottom-underglow-alps-mx)
```
Comment (noroadsleft) [inline keyboards/kprepublic/jj4x4/readme.md]:
```suggestion
Hardware Availability: [KPrepublic](https://kprepublic.com/products/jj4x4-jj4x4-16-keys-custom-mechanical-keyboard-pcb-programmed-numpad-layouts-bface-firmware-with-rgb-bottom-underglow-alps-mx); [AliExpress](https://www.aliexpress.com/item/jj4x4-jj4X4-16-keys-Custom-Mechanical-Keyboard-PCB-programmed-numpad-layouts-bface-firmware-with-rgb-bottom/32901955446.html)
```

File: keyboards/kprepublic/jj4x4/readme.md [hunk 8]
```diff
@@ -6,11 +6,11 @@ A  4x4 keypad kit made and KPRepublic on AliExpress. This is a chopped off versi
...
 
 Make example for this keyboard (after setting up your build environment):
 
-    make jj4x4:default:program
+    make kprepublic/jj4x4:default:program
 
 See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (noroadsleft) [inline keyboards/kprepublic/jj4x4/readme.md]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/kprepublic/jj50/readme.md [hunk 9]
```diff
@@ -10,18 +11,17 @@ Note that this is a complete replacement for the firmware, so you won't be
 using Bootmapper Client to change any keyboard settings, since not all the
 USB report options are supported.  
 
+Keyboard Maintainer: [Wayne K. Jones](github.com/WarmCatUK).  
 Hardware Supported: JJ50 with the ATmega32a chip.  
-Hardware Availability: [KPrepublic on AliExpress](https://www.aliexpress.com/item/jj50-v1-0-Custom-Mechanical-Keyboard-50-PCB-programmed-50-preonic-layouts-bface-firmware-with-rgb/32848915277.html)
-
-This version by [Wayne K. Jones](github.com/WarmCatUK).
+Hardware Availability: [KPrepublic](https://kprepublic.com/collections/jj50-50/products/jj50-50-custom-keyboard-pcb-similar-with-preonic)
```
Comment (noroadsleft) [inline keyboards/kprepublic/jj50/readme.md]:
```suggestion
Hardware Availability: [KPrepublic](https://kprepublic.com/collections/jj50-50/products/jj50-50-custom-keyboard-pcb-similar-with-preonic); [AliExpress](https://www.aliexpress.com/item/jj50-v1-0-Custom-Mechanical-Keyboard-50-PCB-programmed-50-preonic-layouts-bface-firmware-with-rgb/32848915277.html)
```

File: keyboards/kprepublic/cospad/readme.md (hunk 2 above)
Comment (zvecr) [reply keyboards/kprepublic/cospad/readme.md]: Good idea with including both, the main reason was one of the links was tinyurl and asked me to log in.

Comment (noroadsleft) [reply keyboards/kprepublic/cospad/readme.md]: Good spot on that. I tested the old link too and got the same thing.

Perhaps I should change my AliExpress password now... :eyes:

File: keyboards/kprepublic/jj50/readme.md (hunk 9 above)
Comment (zvecr) [reply keyboards/kprepublic/jj50/readme.md]: Done

File: keyboards/kprepublic/jj4x4/readme.md (hunk 8 above)
Comment (zvecr) [reply keyboards/kprepublic/jj4x4/readme.md]: Done

File: keyboards/kprepublic/jj4x4/readme.md (hunk 7 above)
Comment (zvecr) [reply keyboards/kprepublic/jj4x4/readme.md]: Done

File: keyboards/kprepublic/jj40/readme.md (hunk 6 above)
Comment (zvecr) [reply keyboards/kprepublic/jj40/readme.md]: Done

File: keyboards/kprepublic/jj40/readme.md (hunk 5 above)
Comment (zvecr) [reply keyboards/kprepublic/jj40/readme.md]: Done

File: keyboards/kprepublic/daisy/readme.md (hunk 4 above)
Comment (zvecr) [reply keyboards/kprepublic/daisy/readme.md]: Done

File: keyboards/kprepublic/daisy/readme.md (hunk 3 above)
Comment (zvecr) [reply keyboards/kprepublic/daisy/readme.md]: Done

File: keyboards/kprepublic/cospad/readme.md (hunk 1 above)
Comment (zvecr) [reply keyboards/kprepublic/cospad/readme.md]: Done

File: keyboards/kprepublic/cospad/readme.md (hunk 2 above)
Comment (zvecr) [reply keyboards/kprepublic/cospad/readme.md]: I didn't get any interaction with LastPass when i visited the link, go i guess it should be ok. No idea if it had a referral link.

**Comment by zvecr** [comment]
@noroadsleft Thanks for digging out all those AliExpress links.

**Comment by zvecr** [comment]
I have just noticed that on the [kprepublic aliexpress store](https://kprepublic.aliexpress.com/store/group/KTEC-Series/3034003_511912845.html), the list all the boards as **KTEC Series**. Would this be a better folder name to use, for the devices it lists?

**Comment by noroadsleft** [comment]
I think `kprepublic` is better to use than `ktec_series` or a variant, due to name recognition.

**Comment by zvecr** [comment]
Waiting on #5219 so not to break programming behaviour, before continuing. Otherwise, i will revert the changes to the VENDOR_ID and PRODUCT_ID (and hopefully reintroduce the change at a later date).

**Comment by drashna** [comment]
Yikes. Lots of merge conflicts here.  If you could, please update

**Comment by zvecr** [comment]
@drashna Some of the merge conflicts are my fault, and i was waiting for some of my other PRs to land. In that timeframe, a few more boards have landed, i can move those too, take out XD boards, whatever will get this PR across the line fastest.

**Comment by drashna** [comment]
Well, @mechmerlin was the one with a real concern here. 

Personally, If KPRepublic is just selling them, I'm fine with them being in the folder, as long as it's not clear who the actual manufacturer is.

**Comment by drashna** [comment]
There are a bunch of merge conflicts here.  And I think that some of this may have already been done?

**Comment by zvecr** [comment]
@drashna doesnt look like any of the boards have been collated yet, but given the amount of boards being touched, I think i would now want to drive it through the breaking change process.

**Comment by noroadsleft** [comment]
@zvecr I've switched your base branch to `future` to drive this through the breaking change process.

**Comment by fauxpark** [comment]
The BM16A, BM16S and BM43A might also belong here. The designer seems to be "BM Workshop" (nothing on Google) but KPrepublic and Drop are the main places that sell them, apparently.

**Comment by GreatWizard** [comment]
This PR do too much things, We should split it in simple PR in order to be reviewed quickly.
Right now I make a PR that only move boards in the kprepublic directory and adapt make command references in order to works: #9514 

After that we can get the rest of changes into smaller PR?

**Comment by GreatWizard** [comment]
Doucmentation (and its translations) `newbs_flashing.md` have a reference to a make command for the jj40 board that must be updated too

**Comment by GreatWizard** [comment]
Base on my last search, here is the latest list of the KPrepublic boards:

    

- bm16a
-     bm16s
-     bm43a
-     bm60rgb
-     cospad
-     daisy
-     jj4x4
-     jj40
-     jj50
-     staryu
-     xd0002
-     xd0004
-     xd60
-     xd68
-     xd75
-     xd84
-     xd87
-     xd96
-

**Pull request closed without merging**